### DISCUSSION TITLE
cleanup search.cpp/search.h

### DISCUSSION THREAD

**Discussion by Disservin** [official-stockfish/Stockfish#4227] [General]
So while the most work is basically done in search.cpp and while its also a file that many beginners are first directed towards it makes sense to clean this up a bit.

In my opinion the content of search.cpp should be sorted a) by importance
seeing the main functions (our iterative deepening and search<>() ) higher up would ease some work.

I did some basic work on that topic and I'd like to hear other peoples thoughts, for now I just ran Google format over search to that the indentations are somewhat correct. This can be fixed later.

https://github.com/Disservin/Stockfish/blob/search-cleanup/src/search.cpp
https://github.com/Disservin/Stockfish/blob/search-cleanup/src/search.h

I am open towards making this cleanup a joint effort and merging search cleanups into that branch.

**Comment by joergoster** [comment]
Thank you for this.
I took some of the proposed changes and implemented them in my [huntsman2](https://github.com/joergoster/Stockfish/tree/huntsman2) branch.

**Comment by dav1312** [comment]
What do you think? @vondele
